lib/cn/kblock_builder.c

```diff
@@ -1287,6 +1290,131 @@ kbb_set_merge_stats(struct kblock_builder *bld, struct cn_merge_stats *stats)
     bld->mstats = stats;
 }
 
+static merr_t
+kblock_copy_range(
+    struct kblock_desc   *kbd,
+    const struct key_obj *start, /* exclusive */
+    const struct key_obj *end,   /* inclusive */
+    uint64_t             *kb_out)
+{
+    struct wbti *wbti = NULL;
+    struct kblock_builder *kbb;
+    struct key_obj cur = { 0 };
+    const void *kmd;
+    merr_t err;
+    uint64_t tot_keys = 0;
+
+    /* start == NULL implies a copy from the first key
+     * end == NULL implies a copy until EOF
+     */
+    err = kbb_create(&kbb, kbd->cn, NULL);
+    if (err)
+        return err;
+
+    if (start) {
+        char kbuf[HSE_KVS_KEY_LEN_MAX];
+        struct kvs_ktuple seek;
+        uint klen;
+
+        key_obj_copy(kbuf, sizeof(kbuf), &klen, start);
+        kvs_ktuple_init_nohash(&seek, kbuf, klen);
+
+        err = wbti_create(&wbti, kbd->kd_mbd, kbd->kd_wbd, &seek, false, false);
+    } else {
+        err = wbti_create(&wbti, kbd->kd_mbd, kbd->kd_wbd, NULL, false, false);
+    }
+
+    while (!err && wbti_next(wbti, &cur.ko_sfx, &cur.ko_sfx_len, &kmd)) {
+        struct kbb_key_stats stats = { 0 };
+        size_t off = 0, kmd_cnt_off;
+        uint nvals;
```

Comment:
I think you can move these vars into next block?

Reply:
yes, you're right. Fixed.